compiler+runtime/src/cpp/jank/c_api.cpp

```diff
@@ -817,6 +817,42 @@ extern "C"
     return to_hash(o_obj);
   }
 
+  native_integer to_integer(object const *o)
+  {
+    if(o->type == object_type::integer)
+    {
+      return dyn_cast<obj::integer>(o)->data;
+    }
+
+    return to_hash(o);
+  }
+
+  jank_native_integer jank_to_integer(jank_object_ptr const o)
+  {
+    auto const o_obj(reinterpret_cast<object *>(o));
+    return to_integer(o_obj);
+  }
+
+  jank_native_integer shift_mask_case_integer(jank_object_ptr const o,
+                                              jank_native_integer const shift,
+                                              jank_native_integer const mask)
+  {
+    auto const o_obj(reinterpret_cast<object *>(o));
+    auto integer{ to_integer(o_obj) };
+    if(mask != 0)
+    {
+      if(o_obj->type == object_type::integer)
+      {
+        integer = integer >= std::numeric_limits<int32_t>::min()
```

Comment:
Please add parens around the condition of this ternary. Also, please a comment explaining what this fn is doing, what this conditional is doing, and, most importantly, why it's needed.

Reply:
updated.